sqlalchemy_filters/models.py

```diff
@@ -152,13 +187,21 @@ def auto_join(query, *model_names):
     """
     # every model has access to the registry, so we can use any from the query
     query_models = get_query_models(query).values()
-    model_registry = list(query_models)[-1]._decl_class_registry
+    last_model = list(query_models)[-1]
+    model_registry = (
+        last_model._decl_class_registry
+        if sqlalchemy_version_cmp('<', '1.4')
+        else last_model.registry._class_registry
+    )
 
     for name in model_names:
         model = get_model_class_by_name(model_registry, name)
-        if model not in get_query_models(query).values():
+        if model and (model not in get_query_models(query).values()):
             try:
-                query = query.join(model)
+                tmp = query.join(model)
+                if sqlalchemy_version_cmp('>=', '1.4'):  # pragma: nocover
+                    tmp._compile_state()
```

Comment:
In SQLAlchemy >= 1.4 'Select' object has no attribute '_compile_state', so I suppose here the following should be 
```
if sqlalchemy_version_cmp('<', '1.4'):  # pragma: nocover
    tmp._compile_state()
```

Reply:
the compiling state is there for a reason (https://docs.sqlalchemy.org/en/14/changelog/migration_14.html#many-core-and-orm-statement-objects-now-perform-much-of-their-construction-and-validation-in-the-compile-phase) and it works properly in other cases, but ..

... I guess that the test I've added does not trigger auto_join so I don't see it's actually still a bit broken for select object

Reply:
hopefully 2696d7c82c06d0e90dfaa3406db3bc9860e2a9b2 would fix the things, @ArkhipovNikita ?

Reply:
Yes, now everything is great.